File: src/server/backend/runtime/plugins/llm_after/qwen3guard.py
"""Qwen3Guard plugin for LLM output events."""
from __future__ import annotations

from shared.schemas.events import EventType, RuntimeEvent

from backend.runtime.plugins.common.patterns import text_of
from backend.runtime.plugins.qwen3guard import Qwen3GuardPluginBase
from backend.runtime.plugins.registry import register


@register(
    name="qwen3guard_output",
    description="Classify LLM output with Qwen3Guard after model execution.",
)
class Qwen3GuardOutputPlugin(Qwen3GuardPluginBase):
    event_types = [EventType.LLM_OUTPUT]
    policy_scope = "llm_output"

    def _messages_for_event(self, event: RuntimeEvent) -> list[dict[str, str]]:
        content = text_of(event.payload.output)
        return [{"role": "assistant", "content": content}] if content else []

File: src/client/js/agentguard/plugins/llm_after/qwen3guard.js
"use strict";

const { BasePlugin } = require("../base");
const { EventType } = require("../../schemas/events");
const { register } = require("../registry");
const { checkWithQwen3Guard, textOf } = require("../common/qwen3guard");

class Qwen3GuardOutputPlugin extends BasePlugin {
  constructor(options = {}) {
    super(options);
    this.event_types = [EventType.LLM_OUTPUT];
    this.policy_scope = "llm_output";
  }

  check(event) {
    const content = textOf(event.payload?.output);
    const messages = content ? [{ role: "user", content }] : [];
    return checkWithQwen3Guard(this, messages, this.policy_scope);
  }
}

register(
  "qwen3guard_output",
  "Classify LLM output with Qwen3Guard after model execution.",
)(Qwen3GuardOutputPlugin);

module.exports = {
  Qwen3GuardOutputPlugin,
};
